data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ContainsSubstringExpressionFunction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+import java.util.List;
+import javax.inject.Named;
+import java.util.function.Function;
+
+@Named
+public class ContainsSubstringExpressionFunction implements ExpressionFunction {
```

Comment:
I put this question on the issue this PR is for. but wanted to repeat this here: 

I think we should make a decision on how'd we want our expressions to look long term now. Do we want common methods or specific type methods. For example: Do we want to support a single function called `contains` which can support only `strings` and `jsonPointers` referencing `strings` in this first iteration.  But then can be extended to support other data types like `sets`, `maps`, etc. in future revisions. Or do we want to have separate functions for  different data types? I see the benefits and draw backs of both. Have we decided on how we will approach this?

Reply:
We discussed this in a meeting and agreed to rename the function to `contains` to potentially extend the function to other types in future.